hazelcast-client/src/main/java/com/hazelcast/client/config/YamlClientFailoverConfigBuilder.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.config.AbstractYamlConfigBuilder;
+import com.hazelcast.config.ConfigLoader;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.config.yaml.YamlDomChecker;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.internal.yaml.YamlLoader;
+import com.hazelcast.internal.yaml.YamlMapping;
+import com.hazelcast.internal.yaml.YamlNode;
+import com.hazelcast.nio.IOUtil;
+import com.hazelcast.util.ExceptionUtil;
+import org.w3c.dom.Node;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Properties;
+
+import static com.hazelcast.config.yaml.W3cDomUtil.asW3cNode;
+
+/**
+ * Loads the {@link com.hazelcast.client.config.ClientFailoverConfig} using YAML.
+ */
+public class YamlClientFailoverConfigBuilder extends AbstractYamlConfigBuilder {
+
+    private final InputStream in;
+
+    public YamlClientFailoverConfigBuilder(String resource) throws IOException {
+        URL url = ConfigLoader.locateConfig(resource);
+        if (url == null) {
+            throw new IllegalArgumentException("Could not load " + resource);
+        }
+        this.in = url.openStream();
+    }
+
+    public YamlClientFailoverConfigBuilder(File file) throws IOException {
+        if (file == null) {
+            throw new NullPointerException("File is null!");
+        }
+        this.in = new FileInputStream(file);
+    }
+
+    public YamlClientFailoverConfigBuilder(URL url) throws IOException {
+        if (url == null) {
+            throw new NullPointerException("URL is null!");
+        }
+        this.in = url.openStream();
+    }
+
+    public YamlClientFailoverConfigBuilder(InputStream in) {
+        this.in = in;
+    }
+
+    /**
+     * Loads the client failover config using the following resolution mechanism:
+     * <ol>
+     * <li>first it checks if a system property 'hazelcast.client.failover.config' is set. If it exist and
```

Comment:
Can this property be also an xml file name? How do we handle the difference?

Reply:
No, the YAML builder works with YAML configuration only.

Reply:
Clarified in the doc that only `.yaml` files are loaded from system props.